adapters/repos/db/index.go

```diff
@@ -585,25 +586,38 @@ func (i *Index) IncomingBatchAddReferences(ctx context.Context, shardName string
 
 func (i *Index) objectByID(ctx context.Context, id strfmt.UUID,
 	props search.SelectProperties, additional additional.Properties,
+	replProps *additional.ReplicationProperties,
 ) (*storobj.Object, error) {
 	shardName, err := i.shardFromUUID(id)
 	if err != nil {
 		return nil, err
 	}
 
-	local := i.getSchema.
-		ShardingState(i.Config.ClassName.String()).
-		IsShardLocal(shardName)
-
-	if !local {
-		remote, err := i.remote.GetObject(ctx, shardName, id, props, additional)
-		return remote, err
-	}
+	var (
+		obj   *storobj.Object
+		local = i.getSchema.
+			ShardingState(i.Config.ClassName.String()).
+			IsShardLocal(shardName)
+	)
 
-	shard := i.Shards[shardName]
-	obj, err := shard.objectByID(ctx, id, props, additional)
-	if err != nil {
-		return nil, errors.Wrapf(err, "shard %s", shard.ID())
+	if local {
+		if i.replicationEnabled() {
+			obj, err = i.replicator.GetObject(ctx, shardName, id, props, additional, replProps)
```

Comment:
Actually i.replicator.NodeObject() is put in replicator just for convenience. It is not replicating anything and it is there just for debugging.

I find this way more clear but It will be ok with me if you want to keep this version

Reply:
Ok, no problem, I will go with your suggestion then 👍